src/controllers/project.controller/getProject/recommender/index.js

```diff
@@ -0,0 +1,87 @@
+const { TfIdf } = require('natural');
+const { createUserProfile } = require('./userProfile');
+const { createProjectProfile } = require('./projectProfile');
+const { calculateCosineSimilarity } = require('./vectorBasedMapping');
+
+const recommend = async (user, projects) => {
+  // ------------------------------------
+  // Step 1: Prepare user profile
+  // ------------------------------------
+  const userProfile = {
+    profile: await createUserProfile(user),
+  };
+
+  // ------------------------------------
+  // Step 2: Prepare project profiles
+  // ------------------------------------
+  const projectProfiles = new Map();
+  for (let i = 0; i < projects.length; i += 1) {
+    const profile = await createProjectProfile(projects[i]); // Parallelize
```

Comment:
Use Promise.all to calculate project profiles parallely

Reply:
My bad, createProjectProfile isn't an async function, lets remove await & see how we can parallelize